### PR TITLE
#Task_MySQL. Andrey Yakshuk

### PR DESCRIPTION
**Pull request by andreyPromaster**
https://travis-ci.com/github/andreyPromaster/db-assignments/builds/178767332

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -67,7 +84,23 @@ async function task_1_3(db) {
...
+                        COUNT(*)
+                    FROM
+                        Orders) * 100,
+                5) AS "% of all orders"
+    FROM
+        Orders
+            JOIN
+        Customers ON Customers.CustomerID = Orders.CustomerID
+    GROUP BY Customers.CustomerID
+    ORDER BY COUNT(*)/(SELECT COUNT(*) FROM Orders)*100 DESC , Customers.CustomerID ASC;`);
```
Comment (AlexMokin) [inline task/sql-tasks.js]: do not duplicate formula

File: task/sql-tasks.js [hunk 2]
```diff
@@ -91,7 +133,20 @@ async function task_1_5(db) {
...
+    let result = await db.query(`
+    SELECT 
+        ProductName,
+        CategoryName,
+        CompanyName AS 'SupplierCompanyName'
+    FROM
+        Products
+            JOIN
+        Suppliers ON Suppliers.supplierid = Products.supplierid
+            JOIN
```
Comment (AlexMokin) [inline task/sql-tasks.js]: use **inner join** keyword explicitly

File: task/sql-tasks.js [hunk 3]
```diff
@@ -199,7 +322,21 @@ async function task_1_14(db) {
  *
  */
 async function task_1_15(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
```
Comment (AlexMokin) [inline task/sql-tasks.js:325]: try to refator it to this

```
select 
  cont(monthcondition)
From orders
where year = 1997
```

File: task/sql-tasks.js [hunk 4]
```diff
@@ -247,7 +412,22 @@ async function task_1_18(db) {
...
+        SUM(UnitPrice * Quantity) AS 'TotalOrdersAmount, $'
+    FROM
+        Orders
+            JOIN
+        Customers ON Customers.CustomerID = Orders.CustomerID
+            JOIN
+        OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+    GROUP BY Customers.CustomerID
+    HAVING SUM(UnitPrice * Quantity) > 10000
+    ORDER BY SUM(UnitPrice * Quantity) DESC , Orders.CustomerID;`);
```
Comment (AlexMokin) [inline task/sql-tasks.js]: do not duplicate formula

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by andreyPromaster** [comment]
https://travis-ci.com/github/andreyPromaster/db-assignments/builds/181855787

File: task/sql-tasks.js [hunk 5]
```diff
@@ -280,7 +498,34 @@ async function task_1_21(db) {
...
+        t_OrderD.UnitPrice = (SELECT 
+                MAX(t1_OrderD.UnitPrice)
+            FROM
+                Customers AS t1_Cust
+                    JOIN
+                Orders AS t1_Order ON t1_Order.CustomerID = t1_Cust.CustomerID
+                    JOIN
+                OrderDetails AS t1_OrderD ON t1_Order.OrderID = t1_OrderD.OrderID
+            WHERE
+                t1_Cust.CompanyName = t_Cust.CompanyName)
```
Comment (AlexMokin) [inline task/sql-tasks.js]: company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by andreyPromaster** [comment]
https://travis-ci.com/github/andreyPromaster/db-assignments/builds/182417559

**Comment by andreyPromaster** [comment]
Thanks for you review:)

File: task/sql-tasks.js [hunk 6]
```diff
@@ -280,7 +498,35 @@ async function task_1_21(db) {
...
+    FROM
+        Customers AS t_Cust
+            INNER JOIN
+        Orders AS t_Order ON t_Order.CustomerID = t_Cust.CustomerID
+            INNER JOIN
+        OrderDetails AS t_OrderD ON t_OrderD.OrderID = t_Order.OrderID
+            INNER JOIN
+        Products AS t_Prod ON t_OrderD.ProductID = t_Prod.ProductID
+    WHERE
+        t_OrderD.UnitPrice = (SELECT
```
Comment (AlexMokin) [inline task/sql-tasks.js]: still slow. Try to group by customer id and join this subquery on price and cutomer id

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by andreyPromaster** [comment]
https://travis-ci.com/github/andreyPromaster/db-assignments/builds/183346290

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**